src/io/xpi.ts

```diff
@@ -233,4 +237,10 @@ export class Xpi extends IOBase {
       this.zipfile.close();
     }
   }
+
+  _overrideCachedFilesForTests(files: Files) {
```

Comment:
I am not a big fan of exporting a method just for tests... I think I would have preferred to repeat the `myXpi.processed = true;` in all these tests instead, wdyt ?

Reply:
This is future-proofing in case we need to make more changes in the future.. The alternative would be a sub-class for all tests only defined in the test files, maybe? I would like to avoid repeating `.processed = true` everywhere (in addition to `.files = { ... }`).

Reply:
I think I prefer that

Reply:
done